flink-yarn/src/main/java/org/apache/flink/yarn/YarnFlinkApplicationMasterRunner.java

```diff
@@ -158,14 +160,16 @@ protected int runApplicationMaster(Configuration config) {
 				jobManagerRunner.start();
 				LOG.debug("Job Manager Runner started");
 
+				// wait for resource manager to finish and return a value for later to get
+				future = (Future<?>) resourceManager.getTerminationFuture();
+
 				// ---- (5) start the web monitor
 				// TODO: add web monitor
 			}
+			Object object = future.value().get();
 
-			// wait for resource manager to finish
-			resourceManager.getTerminationFuture().get();
 			// everything started, we can wait until all is done or the process is killed
-			LOG.info("YARN Application Master finished");
+			LOG.info("YARN Application Master finished and the result is " + object.toString());
```

Comment:
Just another nitpick: the `object` variable could probably use a nicer / more meaningful name :)